Assets/BoomerangAttack.cs

```diff
@@ -55,7 +55,17 @@ void OnTriggerEnter2D(Collider2D other)
     {
         if (other.gameObject.CompareTag("Enemy"))
         {
-            other.gameObject.GetComponent<EnemyAI>().TakeDamage(damage);
+            EnemyAI enemyAI = other.gameObject.GetComponent<EnemyAI>();
+            if (enemyAI != null)
+            {
+                enemyAI.TakeDamage(damage);
+            }
+
+            FlyingEnemyAI flyingEnemyAI = other.gameObject.GetComponent<FlyingEnemyAI>();
+            if (flyingEnemyAI != null)
+            {
+                flyingEnemyAI.TakeDamage(damage);
+            }
```

Comment:
You might hate me for this... but we should make an Interface class that FlyingEnemyAI and EnemyAI inherit from that has a TakeDamage method. Can still implement this individually, but I would think it's the same for all enemy AI and we can implement it in the interface.